crates/sovereign-sdk/full-node/db/sov-schema-db/src/lib.rs

```diff
@@ -416,6 +441,13 @@ fn default_write_options() -> rocksdb::WriteOptions {
     opts
 }
 
+/// `duration_to_seconds` converts Duration to seconds.
+#[inline]
+pub fn duration_to_seconds(d: Duration) -> f64 {
+    let nanos = f64::from(d.subsec_nanos()) / 1e9;
+    d.as_secs() as f64 + nanos
+}
```

Comment:
IMO nanos is a bit too much precision. We can use micros and just do `d.as_micros() as u64` with no overflow no floating point whatsoever.